pyproject.toml

```diff
@@ -206,7 +206,7 @@ pytest-icdiff = "*"
 robot_descriptions = "*"
 
 [tool.pixi.feature.gpu]
-dependencies = { cuda-version = "12.*", cuda-cupti = "*", jaxlib = "**cuda*" }
+dependencies = { cuda-version = "12.*", cuda-cupti = "*" }
```

Comment:
Any reason for including this change in this PR?

Reply:
`jaxlib = "**cuda*"` gives a TOML parse error. Moreover, it is not necessary to force the `cuda` version since the GPU version of `jaxlib` will be installed automatically if the `__cuda` metapackage is available. 
I tested locally that with that change, the environment produces actually supports GPU:

```sh
$ pixi run -e gpu python                            
 WARN The feature 'style' is defined but not used in any environment
 WARN The feature 'testing' is defined but not used in any environment
 WARN The feature 'viz' is defined but not used in any environment
 WARN The feature 'all' is defined but not used in any environment
Python 3.12.4 | packaged by conda-forge | (main, Jun 17 2024, 10:23:07) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import jax; jax.devices()
[cuda(id=0)]
```

Reply:
Mmh ok. However, sometimes upstream problems / conflicts may propagate downstream preventing the installation of the cuda version. It'd be safer to keep this specification IMO. What about removing one of the leading `*`? Would it be a valid TOML entry?

Reply:
Yes, but it will not find any matching package:

```console
$ pixi run -e gpu python
 WARN The feature 'style' is defined but not used in any envi
ronment
 WARN The feature 'testing' is defined but not used in any en
vironment
 WARN The feature 'viz' is defined but not used in any enviro
nment
 WARN The feature 'all' is defined but not used in any enviro
nment
  × failed to solve the conda requirements of 'gpugroup'
  │ 'linux-64'
  ╰─▶ Cannot solve the request because of: No candidates
      were found for jaxlib cuda.*.
```

Reply:
I see, cc @traversaro

Reply:
As in: https://github.com/Tass0sm/corallab-lib/blob/7c4c201528d2f758c560cb67efdcf2271afd8228/pyproject.toml#L30

Reply:
Last attempt :) If this does not work, no problem. Possible future errors in GPU resolution will be temporary.

```diff
diff --git a/pyproject.toml b/pyproject.toml
index 0699a2c..5298be2 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -206,10 +206,14 @@ pytest-icdiff = "*"
 robot_descriptions = "*"
 
 [tool.pixi.feature.gpu]
-dependencies = { cuda-version = "12.*", cuda-cupti = "*" }
 platforms = ["linux-64"]
 system-requirements = { cuda = "12.1" }
 
+[tool.pixi.feature.gpu.dependencies]
+cuda-version = "12.*"
+cuda-cupti = "*"
+jaxlib = { version = "*", build = "*cuda*" }
+
 [tool.pixi.pypi-dependencies]
 jaxsim = { path = "./", editable = true }
```

Reply:
This works, thanks a lot! Addressed in https://github.com/ami-iit/jaxsim/pull/221/commits/aa532e299c9579665cbb46e6ac477aaff5f3ca54

Reply:
> I'm not sure if pixi in this case expects cuda to be installed system-wide or it installs it from conda-forge.

As for conda, the driver part of cuda should be installed system-wide (and have a newer version w.r.t. to the cuda version used), while the cuda libraries are installed by conda-forge.